File: 06.py
import numpy as np
import re

file = open(r'input\06.txt', 'r')
line = file.readline()
inputdata = np.asarray(re.split(r'\D', line)[:-1], dtype=np.int32)


############################## PART 1 ##############################

currentfish = inputdata
for day in range(80):
    newfish = []
    for fish in currentfish:
        if fish > 0:
            newfish.append(fish - 1)
        else:
            newfish.append(6)
            newfish.append(8)
    currentfish = newfish

print(len(currentfish))

############################## PART 2 ##############################

## setup for c++ test
temp = open(r"input\06_m.txt", "w")
for fish in inputdata:
    temp.write(str(fish) + "\n")
temp.close()

File: 06.cpp
#include <iostream>
#include <vector>
#include <string>
#include <fstream>

using namespace std;

vector<int> getInputdata(string inputfile)
{
    ifstream inputdata;
    inputdata.open(inputfile);
    if (!inputdata)
    {
        cout << "Unable to open file";
        exit(1);
    }

    vector<int> initial;
    string current;

    while (inputdata >> current)
    {
        initial.push_back(stoi(current));
    }
    inputdata.close();

    return initial;
}

vector<int> evening(vector<int> morning)
{
    vector<int> temp;
    for (vector<int>::iterator iter = morning.begin(); iter < morning.end(); iter++)
        {
            if (*iter > 0)
            {
                temp.push_back(*iter - 1);
            }
            else
            {
                temp.push_back(6);
                temp.push_back(8);
            }
        }
    return temp;
}

int main()
{
    vector<int> inputdata = getInputdata(".\\input\\06_m.txt");
    for (int day = 0; day < 150; day++)
    {
        inputdata = evening(inputdata);
        cout << "Completed day " << day << "with current size " << inputdata.size()<< endl;
    }
}